src/main.rs

```diff
@@ -208,14 +208,30 @@ impl Shell {
         pipeline.expand_globs();
         let exit_status = if let Some(command) = commands.get(pipeline.jobs[0].command.as_str()) {
             Some((*command.main)(pipeline.jobs[0].args.as_slice(), self))
-        } else if self.functions.get(pipeline.jobs[0].command.as_str()).is_some() {
-            // Not really idiomatic but I don't know how to clone the value without borrowing self
-            let function = self.functions.get(pipeline.jobs[0].command.as_str()).unwrap().clone();
-            let mut return_value = None;
-            for function_pipeline in function.pipelines.iter() {
-                return_value = self.run_pipeline(function_pipeline, commands)
+        } else if let Some(function) = self.functions.get(pipeline.jobs[0].command.as_str()).cloned() {
+            if pipeline.jobs[0].args.len() - 1 != function.args.len() {
+                println!("This function takes {} arguments, but you provided {}", function.args.len(), pipeline.jobs[0].args.len()-1);
+                Some(NO_SUCH_COMMAND) // not sure if this is the right error code
```

Comment:
Let's add another error code and pick an arbitrary int for it `INCORRECT_NUMBER_OF_ARGS` or something along those lines.